components/ws-daemon/nsinsider/main.go

```diff
@@ -75,6 +79,79 @@ func main() {
 					return unix.Mount("none", c.String("target"), "", unix.MS_SHARED, "")
 				},
 			},
+			{
+				Name:  "mount-fusefs-mark",
+				Usage: "mounts a fusefs mark",
+				Flags: []cli.Flag{
+					&cli.StringFlag{
+						Name:     "source",
+						Required: true,
+					},
+					&cli.StringFlag{
+						Name:     "target",
+						Required: true,
+					},
+					&cli.StringFlag{
+						Name:     "uidmapping",
+						Required: false,
+					},
+					&cli.StringFlag{
+						Name:     "gidmapping",
+						Required: false,
+					},
+				},
+				Action: func(c *cli.Context) error {
+					target := filepath.Clean(c.String("target"))
+					if target == "/tmp" || target == "/" || strings.Contains(target, ",") {
+						return fmt.Errorf("%s cannot be copied up", target)
+					}
+
+					tmp, err := os.MkdirTemp("", "fuse")
+					if err != nil {
+						return err
+					}
+
+					for _, base := range []string{"u", "w", "m"} {
+						dir := filepath.Join(tmp, base)
+						if err := os.MkdirAll(dir, 0755); err != nil {
+							return err
+						}
+					}
+
+					source := filepath.Clean(c.String("source"))
+					args := []string{
+						fmt.Sprintf("lowerdir=%s,upperdir=u,workdir=w", source),
+					}
+
+					if len(c.String("uidmapping")) > 0 {
+						args = append(args, fmt.Sprintf("uidmapping=%v", c.String("uidmapping")))
+					}
+
+					if len(c.String("gidmapping")) > 0 {
+						args = append(args, fmt.Sprintf("gidmapping=%v", c.String("gidmapping")))
+					}
+
+					cmd := exec.Command(
+						fmt.Sprintf("%v/.supervisor/fuse-overlayfs", c.String("source")),
+						"-o",
+						strings.Join(args, ","),
+						"none",
+						"m",
```

Comment:
Could we mount to `target` directly here? What's the benefit of the intermediate bind mount?